File: spec/python/test_js_signed_right_shift.py
import unittest

from js_signed_right_shift import JsSignedRightShift

class TestJsSignedRightShift(unittest.TestCase):
    def test_js_signed_right_shift(self):
        r = JsSignedRightShift.from_file("src/fixed_struct.bin")

        self.assertEqual(r.should_be_40000000, 0x40000000)
        self.assertEqual(r.should_be_a00000, 0xa00000)

File: spec/lua/test_js_signed_right_shift.lua
local luaunit = require("luaunit")

require("js_signed_right_shift")

TestJsSignedRightShift = {}

function TestJsSignedRightShift:test_js_signed_right_shift()
    local r = JsSignedRightShift:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.should_be_40000000, 0x40000000)
    luaunit.assertEquals(r.should_be_a00000, 0xa00000)
end
